WordPress/Classes/Networking/WordPressClient.swift

```diff
@@ -27,7 +26,7 @@ struct WordPressSite {
         } else {
             return WordPressSite(baseUrl: url, type: .selfHosted(
                 username: try blog.getUsername(),
-                authToken: try blog.getPassword())
+                authToken: try blog.getApplicationToken())
```

Comment:
With this change, `WordPressSite` and `WordPressClient` would only be accessible to WP.com sites and self-hosted sites that's authenticated using the new application password flow. That means in places that store them as properties, they'll be declared as optional, until we entirely move away from XMLRPC.

@jkmassel Let me know if that deviates from how you originally designed them or you have any concerns.

Reply:
> With this change, WordPressSite and WordPressClient would only be accessible to WP.com sites and self-hosted sites that's authenticated using the new application password flow.

This is accurate 

> That means in places that store them as properties, they'll be declared as optional, until we entirely move away from XMLRPC.

I'm not too worried about the _how_ but I'd really like to limit optionality. That's why I implemented `static func from(blog: Blog) throws -> WordPressSite` – I figure we don't need to store this type just yet, we can derive it from wherever we pass a `Blog` (or some way to get a `Blog`) around. 

Long-term we'll have some SwiftUI hierarchy where we inject the blog at the appropriate level and all subcomponents can just reference it but until then I think this derivation approach should scale pretty well?